### PR TITLE
[WIP] Initial commit to support RN.

### PR DESCRIPTION
**Pull request by bsansouci**
Here's something that "works". I'm not sure you'd like having two different versions of the code because it's very similar. I'll polish this tomorrow morning. I also need to add tests. This flickers when toggling from `isOpened=true` to `isOpened=false`, not sure why yet.

To test out:
- do all the stuff that's in here http://facebook.github.io/react-native/docs/getting-started.html
- change index.ios.js to

``` js
const React = require('react-native');
const Collapse = require('react-collapse');

const {
  Text,
  View,
} = React;

const Demo = React.createClass({
  getInitialState() {
    return {open: false, text: someText}
  }
  render: function() {
    return (
      <View>
        <Collapse isOpened={this.state.open}>
          <View>
            <Text
              numberOfLines={this.state.open ? null : 2}
              style={{fontSize: 14}}>
              <Text> {this.state.text}</Text>
            </Text>
          </View>
        </Collapse>
        <Text onPress={this.onPress}>Press me</Text>
      </View>
    );
  },

  onPress() {
    this.setState({open: !this.state.open});
  }
});

const someText = `asdf asff as fads dash asf ads fads 
dash adsfadsfadsfds fads fad dash asd fadsf sdaf asdf 
asdf sadf asdf asdf sadf asdf dsafsdaf saf dsa fsaaf f
asdf asff as fads dash asf ads fads 
dash adsfadsfadsfds fads fad dash asd fadsf sdaf asdf 
asdf sadf asdf asdf sadf asdf dsafsdaf saf dsa fsaaf f`;

React.AppRegistry.registerComponent('AwesomeProject', () => Demo);
```

### CODE REVIEW

**Comment by nkbt** [comment]
Quickly checked with some basic example (dynamic text)

react-colapse-native

react-colapse-native-dynamic-text

File: native/Collapse.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+
+  componentWillReceiveProps({children, fixedHeight}) {
+    if (fixedHeight < 0 && children !== this.props.children) {
+      this.setState({dirty: true});
+    }
+  },
+
+
+  shouldComponentUpdate(nextProps, nextState) {
```
Comment (nkbt) [inline native/Collapse.js]: While testing I used 

```
import {shouldComponentUpdate} from 'react-addons-pure-render-mixin';
```

Same as for Web Collapse.

Works pretty well.

**Comment by nkbt** [comment]
I think it would be nice to have generic Collapse component that will simply call separate methods to render children within react-motion. And choose one depending on if it is RN or Web. 

Sam thing for Height reporter. The difference is only in render method.

File: native/Collapse.js (hunk 1 above)
Comment (bsansouci) [inline native/Collapse.js]: Yeah that was a mistake. Fixed.

**Comment by bsansouci** [comment]
It would be nice. But I'm not sure we can easily modularize `HeightReporter` because the web version needs `componentDidUpdate` and a different render, the RN version needs the prop `onLayout` and another render. In the end, those two things do most of the work, so I'm not sure we can abstract enough logic out for it to be worth it. `Collapse` on the other hand could be shared but it would need `React` and `View`/`div`.

**Comment by nkbt** [comment]
HeightReporter probably not. It is quite small and extra abstraction will just hurt. I reckon I could just release it as a separate library. Collapses are almost equal in contrast, I'm going to make and update today to share as much as possible.

**Comment by bsansouci** [comment]
I liked the way react-motion does it with just a function that takes whatever differs between R and RN. Here are the things that will differ:
- https://github.com/nkbt/react-collapse/pull/8/files#diff-a1327b7231704350b7f3560faf4adc1aR103
- View vs div
- React itself

Also why are you storing the height as a string? Why not just a float and have a function to compare them with an epsilon?

**Comment by nkbt** [comment]
This is exactly what I was going to do. Export factory function

**Comment by nkbt** [comment]
Never got a chance to use react native, so cannot really support it...

**Pull request closed without merging**